src/PositionManager.sol

```diff
@@ -155,6 +156,15 @@ contract PositionManager is IPositionManager, ERC721Permit, PoolInitializer, Mul
         // the sender is the payer or receiver
         if (currencyDelta < 0) {
             currency.settle(poolManager, sender, uint256(-int256(currencyDelta)), false);
+
+            // if there are native tokens left over after settling, return to sender
```

Comment:
I feel like this code should live in a sweep function, and it should be on the caller to call sweep? thats what we do in UR

thoughts @hensha256 ?

Reply:
from offline chats: we agreed to do it automatically with an internal function

the majority of operations will need to overpay the native token to account for slippage / price movements